### PR TITLE
fix: Divergence on toolType/toolName naming convention

### PR DESCRIPTION
**Pull request by brunoalvesdefaria**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
  * Bug fix with divergent `toolType`/`toolName` being added to event data.
  * Updated documentation with `toolName` as being the new attribute to be used.
  * Added comments to the code with deprecation notice of `toolType` attribute.

* **What is the current behavior?** (You can also link to an open issue here)
  * Some tools/manipulators were triggering events with `toolType` attribute and others with `toolName` attribute.

* **What is the new behavior (if this is a feature change)?**
  * Until the `toolType` attribute is removed, all the event data are including both `toolType` and `toolName` attributes.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
  * No, but it introduce a deprecation notice for the usage of `toolType` attribute.
  * It will be a breaking change when `toolType` is removed for good.

* **Other information**:
  * `toolType` naming is now going to be used to categorize types of tools (e.g. `general`, `annotation`, `stack`, `segmentation`, etc.)

### CODE REVIEW

File: src/eventListeners/touchEventListeners.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ let lastScale = 1.0,
 const pressDelay = 700,
   pressMaxDistance = 5;
 
-const toolType = 'touchInput';
+const toolName = 'touchInput';
```
Comment (dannyrb) [inline src/eventListeners/touchEventListeners.js:26]: I'm not convinced this is supposed to be `toolName`. "touchInput" appears to be specific to the interaction that's happening here, this is taking place in the `touchEventListener`

File: src/stateManagement/stackSpecificStateManager.js [hunk 2]
```diff
@@ -11,11 +11,11 @@ import {
  * @constructor newStackSpecificToolStateManager
  * @memberof StateManagement
  *
- * @param  {string[]} toolTypes       The tool types to apply to the stack.
+ * @param  {string[]} toolNames       The tool names to apply to the stack.
```
Comment (dannyrb) [inline src/stateManagement/stackSpecificStateManager.js:14]: This description is odd, I had to double check usage, but `toolNames` does appear to be what we want here.

File: src/eventListeners/touchEventListeners.js (hunk 1 above)
Comment (brunoalvesdefaria) [reply src/eventListeners/touchEventListeners.js]: Renamed to `inputName` as it's not a tool.
Also added some `TODO`s as it's calling `setToolOptions` to store `hammer.js` options.

File: src/stateManagement/stackSpecificStateManager.js (hunk 2 above)
Comment (brunoalvesdefaria) [reply src/stateManagement/stackSpecificStateManager.js]: Updated the jsDoc giving a better explanation of this parameter.

**Comment by mikehazell** [comment]
This PR resolved #991

**Pull request merged**